src/rgw/rgw_op.cc

```diff
@@ -5279,29 +5279,38 @@ void RGWRestoreObj::execute(optional_yield y)
     ldpp_dout(this, 1) << "failed to fetch get_obj_attrs op ret = " << op_ret << dendl;
     return;
   }
-  rgw::sal::Attrs attrs = s->object->get_attrs();
-  auto attr_iter = attrs.find(RGW_ATTR_MANIFEST);
+  attrs = s->object->get_attrs();
```

Comment:
I feel all the changes in this function need to filled post calling handle_cloud_tier object, then only those will be filled properly. On second thought do we need to fill this here, it can be done in send_reponse right?

Reply:
As per soumya 
>>> Probably its good to read & save these attrs in RGWRestoreObj::execute() before calling handle_cloudtier_obj(). That way we can identify if its the first restore request as attrs will not be set by then.

Reply:
@soumyakoduri